frontend/src/modules/search/withSearch.js

```diff
@@ -0,0 +1,52 @@
+/* @flow */
+
+import * as React from 'react';
+import mark from 'react-content-marker';
+
+
+export function highlightSearch(base: string, search: string){
+    // Split search string on spaces except if between non-escaped quotes.
+    const unusable = '☠';
+    const searchWords = search.replace(/\\"/g, unusable).match(/[^\s"]+|"[^"]+"/g);
+
+    if (searchWords) {
+        const reg = new RegExp(unusable, 'g');
+        var i = searchWords.length;
+
+        while(i--) {
+            searchWords[i] = searchWords[i].replace(/^["]|["]$/g, '');
+            searchWords[i] = searchWords[i].replace(reg, '"');
+        }
+
+        // Sort array in decreasing order of string length
+        searchWords.sort(function(a, b) {
+            return b.length - a.length;
+        });
+
+        return searchWords.map(function(searchWord) {
+            return mark(
+                searchWord,
+                new RegExp(searchWord, 'i'),
+                searchWord => <mark className='search'>{ searchWord }</mark>,
+            );
+        });
```

Comment:
This is returning an array of functions (of type `ContentMarker`) for a reason I don't understand.

Reply:
@adngdb I don't understand why https://github.com/mozilla/react-content-marker/blob/master/src/mark.js would return `ContentMarker`. Any hints?